GUI/src/ImpactTPlot.py

```diff
@@ -334,54 +340,145 @@ def make4orderPlot(self):
+                                     command=self.plot)
+        self.plot_button.pack(fill='both', expand=1, side='left')
+    def create_figure(self):
+        """Creates the main figure section which will later hold the plot"""
         self.fig = Figure(figsize=(7,5), dpi=100)
         self.subfig = self.fig.add_subplot(111)
-
+    def create_canvas(self):
+        """Create and draw the canvas object"""
         self.canvas = FigureCanvasTkAgg(self.fig, self)
         self.canvas.show()
-        self.canvas.get_tk_widget().pack(side=tk.BOTTOM, fill=tk.BOTH, expand=True)
-    
+        self.canvas.get_tk_widget().pack(side=tk.BOTTOM, 
+                                         fill=tk.BOTH, 
+                                         expand=True)
+    def create_toolbar(self):
+        """Create the bottom toolbar"""
         self.toolbar = NavigationToolbar2TkAgg(self.canvas, self)
         self.toolbar.update()
         self.canvas._tkcanvas.pack(side=tk.TOP, fill=tk.BOTH, expand=True)
-        
-
-        
-class PlotFrame(tk.Frame):
-    def __init__(self, parent,PlotFileName,xl,yl,labelY):
-        tk.Frame.__init__(self, parent)
-        #LARGE_FONT= ("Verdana", 12)
-        #label = tk.Label(self, font=LARGE_FONT,
-        #                 text='plot '+PlotFileName+
-        #                 ' use '+str(xl)+':'+str(yl))
-        #label.pack(pady=10,padx=10)
+    def get_selected_bunch(self):
+        """Find which bunch is selected in the option frame"""
+        if hasattr(self, "bunch_select"):
+            return self.bunch_select.get()
+        else:
+            return 'All'
+    def get_selected_xaxis(self):
+        """Find which x axis is selected in the option frame"""
+        if hasattr(self, "xaxis_select"):
+            return self.xaxis_select.get()
+        else:
+            return 'z'
+    def get_filelist(self):
+        selected_bunch = self.get_selected_bunch()
+        default_filelist = self.get_default_filelist()
+        if selected_bunch == 'All':
+            return default_filelist
+        else:
+            return self.get_bunch_filelist(default_filelist, 
+                                           int(selected_bunch))
+    def get_bunch_filelist(self, filelist, bunch):
+        for idx, filename in enumerate(filelist):
+            basename = '.'.join(filename.split('.')[0:-1])
+            extension = filename.split('.')[-1]
+            new_extension = str(int(extension) + bunch*1000)
+            new_filename = '.'.join([basename, new_extension])
+            filelist[idx] = new_filename
+        return filelist
+    def get_default_filelist(self):
+        raise NotImplementedError()
+    def get_xaxis_parameters(self, tcol, zcol):
+        selected_xaxis = self.get_selected_xaxis()
+        if selected_xaxis == 't':
+            return [tcol, 'time (s)']
+        else:
+            return [zcol, 'z direction (m)']
 
+class PlotFrame(PlotBaseFrame):

```

Comment:
Other plot frames now inherit from `PlotBaseFrame` to access the new features and reduce duplication.